### PR TITLE
feat: adds cassette expirations

### PR DESCRIPTION
**Pull request by Justintime50**
# Description

Adds a test utility that checks for expired cassettes on each run and throws errors if cassettes must be re-recorded due to old age to ensure the data is always up-to-date within a half year.

# Testing

I set the days to ~40 and errors were thrown, setting it to 180 has the suite pass 

# Pull Request Type

Please select the option(s) that are relevant to this PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement (fixing a typo, updating readme, renaming a variable name, etc)

### CODE REVIEW

**Comment by Justintime50** [comment]
Blocked until #232 is merged

File: tests/conftest.py
Comment (nwithan8) [inline tests/conftest.py:91]: Nit: I prefer to be explicit on parameters, even if it's the same as the default value. Just makes it more readable.

File: tests/conftest.py
Comment (nwithan8) [inline tests/conftest.py:80]: Question: Is there no way to store a timestamp inside the file rather than relying on the file metadata? Moving the file would change the timestamp, even though the contents aren't changed.

**Review by nwithan8 — DISMISSED** [review]

File: tests/conftest.py
Comment (Justintime50) [reply tests/conftest.py]: This is no longer applicable

File: tests/conftest.py
Comment (Justintime50) [reply tests/conftest.py]: I checked the docs for `vcrpy` and there's no mention of a timestamp sadly. They don't include one we could use either.

**Review by jontsai — APPROVED** [review]

**Review by nwithan8 — APPROVED** [review]

**Pull request merged**
